presto-oracle/src/test/java/io/prestosql/plugin/oracle/TestingOracleServer.java

```diff
@@ -36,17 +36,28 @@ public TestingOracleServer()
     {
         super("wnameless/oracle-xe-11g-r2");
 
-        // this is added to allow more processes on database, otherwise the tests end up giving a ORA-12519
-        // to fix this we have to change the number of processes of SPFILE
-        // but this command needs a database restart and if we restart the docker the configuration is lost.
-        // configuration added:
-        // ALTER SYSTEM SET processes=500 SCOPE=SPFILE
-        // ALTER SYSTEM SET disk_asynch_io = FALSE SCOPE = SPFILE
-        this.addFileSystemBind("src/test/resources/spfileXE.ora",
-                "/u01/app/oracle/product/11.2.0/xe/dbs/spfileXE.ora",
-                BindMode.READ_ONLY);
-
         start();
+
+        try (Connection connection = DriverManager.getConnection(getJdbcUrl(), super.getUsername(), super.getPassword());
+                Statement statement = connection.createStatement()) {
+            // this is added to allow more processes on database, otherwise the tests end up giving
+            // ORA-12519, TNS:no appropriate service handler found
+            // ORA-12505, TNS:listener does not currently know of SID given in connect descriptor
+            // to fix this we have to change the number of processes of SPFILE
+            statement.execute("ALTER SYSTEM SET processes=1000 SCOPE=SPFILE");
+            statement.execute("ALTER SYSTEM SET disk_asynch_io = FALSE SCOPE = SPFILE");
+        }
+        catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+
+        try {
+            execInContainer("/bin/bash", "/etc/init.d/oracle-xe", "restart");
```

Comment:
Does it wait for the db to be back opertional?

Reply:
@findepi As far as checked the script, yes. But added `waitUntilContainerStarted()` just in case.